### PR TITLE
[ticket/12574] Remove passwords manager dependency from auth providers

### PR DESCRIPTION
**Pull request by marc1706**
Checklist:

- [x] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-12574

### CODE REVIEW

File: tests/auth/provider_ldap_test.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+				'user_ip'				=> '',
+			),
+		);
+
+		$this->assertEquals($expected, $this->provider->login($username, $password));
+	}
+
+	public function test_autologin()
+	{
+		$this->assertNull($this->provider->autologin());
```
Comment (Derky) [inline tests/auth/provider_ldap_test.php]: Why do you assertNull for the autologin? (Is that not possible?)

File: tests/auth/provider_ldap_test.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	}
+
+	public function test_validate_session()
+	{
+		$user = array(
+			'username'	=> 'admin',
+			'user_type'
+		);
+
+		$this->assertNull($this->provider->validate_session($user));
```
Comment (Derky) [inline tests/auth/provider_ldap_test.php]: Why assertNull here?

File: tests/auth/provider_ldap_test.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	public function test_autologin()
+	{
+		$this->assertNull($this->provider->autologin());
+	}
+
+	public function test_validate_session()
+	{
+		$user = array(
+			'username'	=> 'admin',
+			'user_type'
```
Comment (Derky) [inline tests/auth/provider_ldap_test.php]: I think you can remove this key or set or value.

File: tests/auth/provider_ldap_test.php (hunk 1 above)
Comment (marc1706) [reply tests/auth/provider_ldap_test.php]: Correct, it's not supported.

File: tests/auth/provider_ldap_test.php (hunk 2 above)
Comment (marc1706) [reply tests/auth/provider_ldap_test.php]: It's not supported, hence a user has to re-login once their session has expired.

**Pull request merged**